R/deltaccd.R

```diff
@@ -87,6 +87,45 @@ calcCCDSimple = function(ref, emat, method = 'spearman', scale = FALSE) {
   return(ccd)}
 
 
+checkVar = function(emat, groupVec) {
+  
+  varCheck = foreach (groupNow = sort(unique(groupVec)), .combine = rbind) %do% {
+    
+    varMat = apply(emat[, groupVec == groupNow], MARGIN = 1, 
+                   FUN = stats::var, na.rm = TRUE)
+    varDt = data.table::as.data.table(varMat, keep.rownames = 'gene')
+    data.table::setnames(varDt, 'varMat', 'variance')
+    varDt[, group := groupNow]
+    
+    zeroVar = varDt[variance == 0]
+    zeroVar[, variance := NULL]
+    
+    return(zeroVar)}
+  
+  if (nrow(varCheck) > 0) {
+    stop('Zero variance in the following gene-group pairs:\n', 
+         paste(utils::capture.output(print(varCheck)), collapse = '\n'))}}
+
+
+checkGenes = function(emat, refCor) {
+  geneNames = rownames(refCor)[rownames(refCor) %in% rownames(emat)]
+  if (length(geneNames) < nrow(refCor)) {
+    missingGenes = setdiff(rownames(refCor), geneNames)
+    stop(paste0('The following gene(s) is/are not in the expression matrix: \n',
+                paste0(missingGenes, collapse = '\n')))} 
+  
+  return(geneNames)}
+
+
+checkRefCor = function(refCor, refEmat = NULL, method = 'spearman') {
```

Comment:
Look at the code more carefully. What value should the function return? This is why we need tests.

Reply:
Oops, I think it needs to return `refCor` for the rest of calcCCD/calcDeltaCCD to work with a `refEmat`.